src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java

```diff
@@ -359,12 +359,21 @@ private void executePatchTransactionV2(PaymentRequestEntity entity, String reque
         Long transactionStatus = entity.getStatus().equals(AUTHORIZED.name()) ? TX_AUTHORIZED_BY_PGS.getId() : TX_REFUSED.getId();
         String authCode = entity.getAuthorizationCode();
         PatchRequest patchRequest = new PatchRequest(transactionStatus, authCode);
-        try {
-            String result = restapiCdClient.callPatchTransactionV2(Long.valueOf(entity.getIdTransaction()), patchRequest);
-            log.info(String.format("Response from PATCH updateTransaction for requestId %s is %s", requestId, result));
-        } catch (Exception e) {
-            log.error(PATCH_CLOSE_PAYMENT_ERROR + requestId, e);
+        boolean isToContinue = true;
+        int retryCount = 1;
+        while (isToContinue && retryCount <= MAX_RETRIES) {
```

Comment:
can't you use a `restTemplate` retry mechanism?

Reply:
I used Feign retry mechanism, because it is used instead of restTemplate

Reply:
ok, to do in another PR